src/services/StreamService.ts

```diff
@@ -75,22 +76,23 @@ class StreamService {
         Object.assign(templateRecord, {});
```

Comment:
looking at this function, i dont think there's a need to make the array, you could just validate that there are test types in the array, then do map to look through the test type array and return the record couldn't you?